torch/distributed/_functional_collectives.py

```diff
@@ -375,6 +411,41 @@ def reduce_scatter_tensor(
     res = _maybe_wrap_tensor(tensor)
     return res
 
+
+def all_reduce_coalesced(self: List[torch.Tensor], reduceOp: str, group: RANK_TYPES, tag: str = "") -> List[torch.Tensor]:
+    """
+    Reduces a list of tensors across all machines in such a way that all get
+    the final result.
+
+    The all tensors in the input list are left unmodified.
```

Comment:
Should we mention how wait is performed? That is an important contract.

Reply:
WaitHolder turned out to be not what we need. I fix it in the next PR